Routers/BookingRouter.cs

```diff
@@ -0,0 +1,118 @@
+
+            var remainingBookings = GetAll().Where(booking => booking.BookingId != bookingId).ToList();
+
+            _bookingRepositoty.DeleteTheFileData();
+
+            foreach (var booking in remainingBookings)
+            {
+                _bookingRepositoty.Save(booking);
+            }
+        }
+
+        public List<Booking> GetBookingForPassenger(int passengerPassportId)
+        {
+            return GetAll().Where(booking => booking.PassengerPassportId == passengerPassportId).ToList();
+        }
+
+
+        public List<Booking> FilterByPrice(decimal price)
+        {
+            return GetAll().Where(booking => booking.Price == price).ToList();
+        }
+
+        public List<Booking> FilterByDepartureCountry(string departureCountry)
+        {
+            return GetAll().Where(booking =>
+                         _flightRouter.GetByFlightNumber(booking.FlightId).DepartureCountry
+                         .Equals(departureCountry, StringComparison.InvariantCultureIgnoreCase))
+                         .ToList();
+        }
+
+        public List<Booking> FilterByDestinationCountry(string destinationCountry)
+        {
+            return GetAll().Where(booking =>
+                            _flightRouter.GetByFlightNumber(booking.FlightId).DestinationCountry
+                            .Equals(destinationCountry, StringComparison.InvariantCultureIgnoreCase))
+                            .ToList();
+        }
+
+        public List<Booking> FilterByDepartureDate(DateTime departureDateTime)
+        {
+            return GetAll().Where(booking =>
+                            _flightRouter.GetByFlightNumber(booking.FlightId).DepartureDateTime == departureDateTime)
+                            .ToList();
+        }
+
+        public List<Booking> FilterByDepartureAirport(string departureAirport)
+        {
+            return GetAll().Where(booking =>
+                            _flightRouter.GetByFlightNumber(booking.FlightId).DepartureAirport
+                            .Equals(departureAirport, StringComparison.InvariantCultureIgnoreCase))
+                            .ToList();
+        }
+
+        public List<Booking> FilterByArrivalAirport(string arrivalAirport)
+        {
+            return GetAll().Where(booking =>
+                            _flightRouter.GetByFlightNumber(booking.FlightId).ArrivalAirport
+                            .Equals(arrivalAirport, StringComparison.InvariantCultureIgnoreCase))
+                            .ToList();
+        }
+
+        public List<Booking> FilterByFlightClass(ClassType flightClass)
+        {
+            return GetAll().Where(booking =>
+                            _flightRouter.GetByFlightNumber(booking.FlightId).ClassType == flightClass)
+                            .ToList();
+        }
+
+        public List<Booking> FilterByFlightId(int flightId)
+        {
+            return GetAll().Where(booking => booking.FlightId == flightId).ToList();
+        }
+
```

Comment:
For your interest: the manager might need to filter based on many parameters so we should support filtering based on all the specified parameters in one method. 
You will need to build the filter query manually. so you need to add a where clause to filter based on some parameter if the input for that parameter is not null or empty for example.